packages/optimizer-docker/Dockerfile

```diff
@@ -0,0 +1,12 @@
+from node:lts-alpine3.12
+
+# Layer 1, add package.json and install deps.
+RUN echo "Building amp-optimizer-docker" 
+ADD package.json .
+RUN yarn
+
+# Layer 2: add src files
```

Comment:
In general each Dockerfile command creates a [Layer](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/). You want to stack your layers such that the more frequently changing bits are lower down, since layers are cached and work can be skipped on rebuilds.

When its written like this, a `yarn` command only need to happen when the package.json changes as opposed to on each change to the `index.js` or `utils.js` files.

That said, I should also copy over the lockfile